src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -605,8 +609,16 @@ protected Query doToQuery(QueryShardContext context) {
                 .knnEngine(knnEngine)
                 .indexName(indexName)
                 .fieldName(this.fieldName)
-                .vector(VectorDataType.FLOAT == vectorDataType ? this.vector : null)
-                .byteVector(VectorDataType.BYTE == vectorDataType || VectorDataType.BINARY == vectorDataType ? byteVector : null)
+                .vector(
```

Comment:
Can we simplify here , conditions are getting broader ? Probably we can get requestbuilder first with params which which is common and then setting specific one based on condition.

Reply:
Refactored it by extracting the conditions into 2 separate methods